providers/redis/redis.go

```diff
@@ -287,8 +303,62 @@ func (p *RedisProvider) GetState(ctx context.Context, instanceID string) (provid
 		return providers.ServiceState(""), "", fmt.Errorf("Invalid response from AWS: status is missing for %s", replicationGroupID)
 	}
 
-	message := p.getMessage(ctx, replicationGroup)
+	if timeOut != "" {
+		timeOutParsed, err := time.Parse(time.RFC3339, timeOut)
+		if err != nil {
+			return providers.ServiceState(""), "", errors.Wrap(err, "Failed to parse time out string")
+		}
 
+		if time.Now().Unix() > timeOutParsed.Unix() {
+			return providers.ServiceState(""), "", fmt.Errorf("Operation has timed out for replication group ID %s", replicationGroupID)
+		}
+	}
+
+	if *replicationGroup.Status == "available" {
+
+		if operation == "failover" {
+			primaryNode := ""
+			nodeToFailOverTo := ""
+			for _, nodeGroup := range replicationGroup.NodeGroups {
+				for _, nodeGroupMember := range nodeGroup.NodeGroupMembers {
+					if *nodeGroupMember.CurrentRole == "primary" {
```

Comment:
Are there constants for the values of `CurrentRole` in the AWS SDK?

Reply:
Sadly not. They just get returned as a string ptr.